### ISSUE TITLE
Types are broken as of 6.10.0

### ISSUE THREAD

**Issue by JamiesWhiteShirt** [jbetancur/react-data-table-component#667]
## Issue Check list
- [x] Agree to the [Code of Conduct](https://github.com/jbetancur/react-data-table-component/blob/master/CODE-OF-CONDUCT.md)
- [x] Read the README
- [ ] You are using React 16.8.0+
- [x] You installed `styled-components`
- [ ] Include relevant code or preferably a [code sandbox](https://codesandbox.io/embed/react-data-table-sandbox-ccyuu
)

## Describe the bug
When importing DataTable, the types are invalid, causing TypeScript to fail to compile unless the "skipLibCheck" compiler option is set to true.

```
ERROR in ./node_modules/react-data-table-component/dist/index.d.ts(17,24):
TS2304: Cannot find name 'T'.
```

The changes in #642 did not account for the generic type `T` being used elsewhere in the interfaces.

## To Reproduce
1. Create TypeScript project
2. Install react-data-table-component
3. Import DataTable from react-data-table-component
4. Compile
5. See error

## Expected behavior
The types included with the project should be valid.

## Code Sandbox, Screenshots, or Relevant Code
Unfortunately, I am unable to reproduce the problem on Code Sandbox.

## Versions (please complete the following information)
 - React: Preact 10.4.7, but N/A
 - Styled Components: 5.1.1
 - OS: macOS Catalina 10.15.6
 - Browser: N/A

**Comment by jbetancur** [comment]
I pulled out the generic types a bit prematurely. However, I believe this PR should restore the functionality https://github.com/jbetancur/react-data-table-component/pull/665 once I test/approve/merge it in

**Comment by jbetancur** [comment]
Should be fixed now in 6.11.2 via #665

**Comment by JamiesWhiteShirt** [comment]
I can confirm that it was fixed in 6.11.2.
